custom_components/casambi_bt/__init__.py

```diff
@@ -69,58 +63,50 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     return unload_ok
 
 
-async def async_casmbi_api_setup(
-    hass: HomeAssistant, address: str, password: str
-) -> CasambiApi | None:
-    client = hass.helpers.httpx_client.get_async_client()
-    try:
-        casa = Casambi(client)
-        device = bluetooth.async_ble_device_from_address(
-            hass, address, connectable=True
-        )
-        if not device:
-            raise NetworkNotFoundError
-        await casa.connect(device, password)
-    except BluetoothError as err:
-        raise ConfigEntryNotReady("Failed to use bluetooth") from err
-    except AuthenticationError as err:
-        raise ConfigEntryAuthFailed(
-            f"Failed to authenticate to network {address}"
-        ) from err
-    except NetworkNotFoundError as err:
-        raise ConfigEntryNotReady(
-            f"Network with address {address} wasn't found"
-        ) from err
-    except Exception as err:  # pylint: disable=broad-except
-        raise ConfigEntryError(f"Unexpected error creating network {address}") from err
-
-    api = CasambiApi(casa, hass, address, password)
-    return api
-
-
 class CasambiApi:
     _callback_map: dict[int, list[Callable[[Unit], None]]] = {}
     _cancel_bluetooth_callback: Callable[[], None] = None
     _reconnect_lock = asyncio.Lock()
 
     def __init__(
-        self, casa: Casambi, hass: HomeAssistant, address: str, password: str
+        self, hass: HomeAssistant, address: str, password: str
     ) -> None:
-        self.casa = casa
         self.hass = hass
         self.address = address
         self.password = password
+        self.casa: Casambi = Casambi(hass.helpers.httpx_client.get_async_client())
 
-        self._register_bluetooth_callback()
-
-    def _register_bluetooth_callback(self):
+    def register_bluetooth_callback(self):
```

Comment:
Is there a reason why the _ was removed? As far as I can see this method is still only used internally.

Reply:
I thought this is the convention for internal methonds

Reply:
Ah, right, it is used only internal now again. So will change it back.